lang/rapidjson/Portfile

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem                      1.0
+PortGroup                       github 1.0
+PortGroup                       cmake 1.1
+
+github.setup                    Tencent rapidjson b32cd9421c5e3cbe183a99b6537ce11441e50656
+version                         20180424
+license                         MIT
+categories                      lang devel
+platforms                       darwin
+maintainers                     @Lord-Kamina
+description                     A fast JSON parser/generator for C++ with both SAX/DOM style API.
+long_description                RapidJSON is a fast, unicode-friendly, self-contained \
+                                and header-only library without any dependencies for parsing JSON.
+checksums                       rmd160  a0211ac1dc84270169b59c584894a0f49d0030a8 \
+                                sha256  4133516f8fd578053f478f3b6211ebcfb03056bdd59d6c6d0312c393f467cf37
+homepage                        http://rapidjson.org/
+depends_build-append            bin:doxygen:doxygen
+
+# Disbled Valgrind because it's apparently failing not due to issues with RapidJSON but rather macOS itself.
+patchfiles                      patch-test-CmakeLists.txt.diff \
+                                patch-test-unittest.diff \
+                                patch-test-unittest-CmakeLists.txt.diff
+
+configure.pre_args-delete       DCMAKE_POLICY_DEFAULT_CMP0025=NEW
+configure.post_args-append      -DRAPIDJSON_HAS_STDSTRING=ON -DRAPIDJSON_HAS_CXX11_RVALUE_REFS=ON -DRAPIDJSON_BUILD_TESTS=OFF
+configure.optflags-delete   -   Os
+cmake.generator                 Unix Makefiles
+
+variant tests description {Build and run unit tests.} {
+    depends_build-append        port:gtest
+    configure.post_args-delete  -DRAPIDJSON_BUILD_TESTS=OFF
+    test {
+        test.run                    yes
+        test.cmd                    make test
+        system -W ${cmake.build_dir} ${test.cmd}
+        }
```

Comment:
This cannot work like this. `test.run` has to be defined outside of the phase. Why are you using system here?

You should only need these two options:
```
test.run yes
test.target test
```

Reply:
I did it that way the first time I tried and for some reason it did nothing, where as defining it as I did worked fine.

I'm gonna try it again later.